examples/step-90/doc/intro.dox

```diff
@@ -0,0 +1,95 @@
+                  {(1 + 0.5  \sin(\pi  x))^{2}} =
+                1
+ @f}
+ We note that the surface's curvature varies significantly.
+
+<h3>Model problem</h3>
+
+We would like to solve the simplest possible problem, Laplace--Beltrami equation,
+@f{equation*}
+  -\Delta_\Gamma u + u = f \qquad  \text{in }\, \Gamma,
+@f}
+ Here we added the term $u$ to the left-hand side so the problem becomes well-posed.
+ The force $f$ is calculated analytically for a manufactured solution,
+ @f{equation*}
+ u=xy\,,
+  @f}
+using the exact normal $\mathbf{n}$ and the exact Hessian $\nabla^2\mathbf{n}$ of the surface computed analytically.
+
+<h3>The Trace Finite Element Method</h3>
+
+TraceFEM is an unfitted method meaning the surface $\Gamma$  is immersed into a regular, uniform background mesh that
+ stays fixed even if the surface would be evolving.
+To solve Laplace--Beltrami equation, we first construct an approximation, denoted by $\Gamma_h$,
+of $\Gamma$ in a form of quadrature points for the surface integration combined with the approximate surface normals.
+Next we distribute degrees of freedom over a thin subdomain $\Omega_h$
+that completely covers the $\Gamma_h$ and that consists of the intersected cells $\mathcal{T}_\Omega^h$,
+@f{equation*}
+  \mathcal{T}_\Omega^h = \{ T \in \mathcal{T}^{h} : T \cap \Gamma_h \neq \emptyset \}.
+@f}
+The finite element space where we want to find our numerical solution, $u_h$, is now
+@f{equation*}
+  V_h = \{ v \in C(\Omega_h) : v \in Q_p(T), \, T \in \mathcal{T}_\Omega^h \},
+@f}
+where $\Omega_h$ is the union of all intersected cells from $\bigcup_{T \in \mathcal{T}_\Omega^h} \overline{T}$.
+
+To create $V_h$, we first add an FE_Q and an
+FE_Nothing element to an hp::FECollection. We then iterate over each cell,
+$T$, and depending on whether $T$ belongs to $\mathcal{T}_\Omega^h$ or not,
+we set the active_fe_index to either 0 or 1.
+For this purpose, we will use the class NonMatching::MeshClassifier.
+
+A natural candidate for a weak formulation  involves the following bilinear forms
+@f{align*}
+  a_h(u_h, v_h) =  (\nabla_{\Gamma_h} u_h, \nabla_{\Gamma_h} v_h)_{\Gamma_h}
+                  +(u_h, v_h)_{\Gamma_h}\,,\qquad
+  L_h(v_h)      =  (f^e,v_h)_{\Gamma_h}.
+@f}
+where $f^e$ is an extension of $f$ from $\Gamma$ to $\Omega_h$.
+
+However, the so-called ''small-cut" problem may arise and one should
+introduce the stabilized version of TraceFEM: Find $u_h \in V_h$ such that
+@f{equation*}
+  a_h(u_h,v_h) + s_h(u_h, v_h) = L_h(v_h), \quad \forall v_h \in V_\Omega^h,
+@f}
+Here the normal-gradient stabilization $s_h$ involves the three-dimensional integration over whole (but intersected) cells and is given by
+@f{equation*}
+  s_h(u_h,v_h) = h^{-1}(\mathbf{n}_h\cdot\nabla u_h, \mathbf{n}_h\cdot\nabla v_h)_{\Omega_h},
+@f}
+Note that the $h^{-1}$ scaling may be relaxed for sufficiently smooth solutions such as the manufactured one, but we
+choose the strong scaling to demonstrate the worst case.
```

Comment:
This might be a good place to provide a reference to a paper that explains the origin or reasoning of the stabilization.